pkg/services/pluginsintegration/plugininstaller/service.go

```diff
@@ -166,11 +148,34 @@ func (s *Service) installPlugins(ctx context.Context, pluginsToInstall []setting
 	return nil
 }
 
-func (s *Service) Run(ctx context.Context) error {
-	err := s.installPlugins(ctx, s.cfg.PreinstallPluginsAsync, false)
-	if err != nil {
-		// Unexpected error, asynchronous installation should not return errors
-		s.log.Error("Failed to install plugins", "error", err)
+func (s *Service) starting(ctx context.Context) error {
+	if len(s.cfg.PreinstallPluginsSync) > 0 {
+		s.log.Info("Installing plugins", "plugins", s.cfg.PreinstallPluginsSync)
+		if err := s.installPlugins(ctx, s.cfg.PreinstallPluginsSync, true); err != nil {
+			s.log.Error("Failed to install plugins", "error", err)
+			return err
+		}
+	}
+	s.log.Info("Plugins installed", "plugins", s.cfg.PreinstallPluginsSync)
+	return nil
+}
+
+func (s *Service) running(ctx context.Context) error {
+	if len(s.cfg.PreinstallPluginsAsync) > 0 {
+		s.log.Info("Installing plugins", "plugins", s.cfg.PreinstallPluginsAsync)
+		if err := s.installPlugins(ctx, s.cfg.PreinstallPluginsAsync, false); err != nil {
+			s.log.Error("Failed to install plugins", "error", err)
+			return err
+		}
 	}
+	<-ctx.Done()
 	return nil
 }
+
+func (s *Service) Run(ctx context.Context) error {
```

Comment:
It's possible due to the `IsDisabled()` logic that we may never run here and end up with:

```
Error: ✗ invalid service state: Failed, expected: Terminated, failure: invalid service state: Failed, expected: Running, failure: no such module: plugin.backgroundinstaller
```

Therefore we can update the starting func:
```go
func (s *Service) starting(ctx context.Context) error {
	if len(s.cfg.PreinstallPluginsSync) == 0 {
		<-ctx.Done()
		return nil
	}

	s.log.Info("Installing plugins", "plugins", s.cfg.PreinstallPluginsAsync)
	if err := s.installPlugins(ctx, s.cfg.PreinstallPluginsAsync, false); err != nil {
		s.log.Error("Failed to install plugins", "error", err)
		return err
	}
	<-ctx.Done()
	return nil
}
```

Reply:
BTW I also checked the other services in the map, but there doesn't appear to be any that have logic in their `IsDisabled()` thankfully

Reply:
i think this is handled by a change i made to the manager. i registered a nil module for disabled services. thoughts?